### ISSUE TITLE
Make testing_roc_alloc  align the allocation

### ISSUE THREAD

**Issue by MatthewJohnHeath** [roc-lang/roc#7287]
Currently `testing_roc_alloc`in crates/compiler/builtins/bitcode/src/utils.zig ignores the value passed as alignment and always returns a pointer to memory with alignment 1.  This caused me a few hours of confusion trying to test an allocation that needed aligning.
Ideally, it would align to whatever value is passed (up to that supported by Zig), but allocating with an alignment not known at compile time seems hard (?). 
Always aligning to 16 and panicking if a higher alignment is requested seems like it would cover all likely cases (without wasting too many bytes)

**Comment by Anton-4** [comment]
I started a discussion for this: https://roc.zulipchat.com/#narrow/channel/395097-compiler-development/topic/testing_roc_alloc/near/485604671

**Comment by lukewilliamboswell** [comment]
@MatthewJohnHeath are you planning on implementing this?

**Comment by MatthewJohnHeath** [comment]
@lukewilliamboswell yes,. I don't think I have permissions for making a pull request to the real repo, but I'll try to the changes on a  branch of my forked version today.

**Comment by MatthewJohnHeath** [comment]
Oh... this does not work quite how I thought it did though. It will take longer

**Comment by MatthewJohnHeath** [comment]
Oh it actually is pretty simple, the pointer just needed to be passed to the dealloc as a type with the same alignment : Here is the branch https://github.com/MatthewJohnHeath/rocSHA/tree/7287-Make-testing_roc_alloc-align-the-allocation

**Comment by MatthewJohnHeath** [comment]
Technically it is aligning as whatever @alignOf(usize) is. Making it 8 on arbitrary hardware is probably possible, but it's trickier and there isn't an obvious benefit.

**Comment by MatthewJohnHeath** [comment]
OH wait again! It was meant to be to 16!

**Comment by MatthewJohnHeath** [comment]
OK this is[ the PR](https://github.com/roc-lang/roc/issues/7287) (The git hub instructions to link the issue and the PR are not working for me)

**Comment by JRI98** [comment]
Fixed by https://github.com/roc-lang/roc/pull/7312?

**Comment by MatthewJohnHeath** [comment]
It should be, but I think I stopped a  mistake in my code that was merged there....
